src/components/AllArticles.jsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState, useContext } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { fetchArticles } from "../../utils";
+import ArticleCard from "./ArticleCard";
+import ArticleList from "./ArticleList";
+
+function AllArticles() {
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
```

Comment:
fetching Articles often takes some time - let's add a loading state so the user knows what's going on!